job-descriptions/software-engineer-code-exercise.md

```diff
@@ -0,0 +1,60 @@
+- You may use open-source libraries, but most candidates don't find it necessary.
+- After two hours, you will email us your solution as a zip file and we will get back to you on next steps within 2 business days.
+
+If we decide to move forward, we will schedule a 30-minute followup call to discuss your code.
+
+## Go code exercise
+
+We designed this exercise to measure your understanding of HTTP requests, concurrency, and error handling.
+
+- You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time that works for you; there are no scheduling constraints.
+- At your chosen date and time, we will send you detailed instructions. You will be building a simple API server in Go that calls a JSON endpoint of another service and exposes a JSON endpoint of its own.
+- You should have a recent version of Go installed on your computer.
+- You can look up documentation on the internet while you are coding.
+- You are not prohibited from using open-source libries, but the problem is easily solved using only the Go standard library.
+- After two hours, you will email us your solution as a zip file and we will get back to you on next steps within 2 business days.
+
+If we decide to move forward, we will schedule a 30-minute followup call to discuss your code.
+
+## FAQ
+
+### Why do you do a coding exercise?
+
+A significant part of your job as a software engineer is to write code, so we naturally want to measure your ability here during our interview process.
+
+An asynchronous coding exercise provides the most realistic environment for measuring how you would work at Sourcegraph on a daily basis.
+
+- You can use the programming environment that you are most familiar with (i.e. your own laptop with your own editor with all the keyboard shortcuts and extensions/tools that you like, and with access to documentation on the internet).
+- You don't have the extra pressure of coding while someone else is watching.
+- You can schedule our asynchronous coding exercise for the best time that works for you.
+
+Candidates who have completed our coding exercise regularly tell us that they enjoyed it and/or learned something.
+
+### Why a 2-hour limit?
+
+We choose to enforce a finite limit so that we get a consistent signal across all candidates. It wouldn't be fair if candidates could spend as much time as they wanted, because some candidates have more time than others.
+
+2 hours is long enough that we can ask non-trivial questions that test for relevant skills while also being short enough that it shouldn't be an unreasonable burden for anyone.
+
+### What if I have significant open-source contributions?
+
+Many engineers on our team have made significant open-source contributions, so we view that as a strong positive signal on your application and we will probably ask you about your contributions during interviews; however, we still ask all candidates to complete our coding exercise.
```

Comment:
This may be something to consider separately.

I suggest we also say that, while significant open-source contributions are a strong positive, _lack_ of significant open-source contributions is not (in itself) a negative. A lot of talented folks simply haven't had the time or opportunity to get deeply hooked into a popular open-source project. I think it's good for us to be explicit so that folks who haven't had those opportunities won't be put off from applying.

Reply:
To avoid sending the wrong signals, I just eliminated the implication that open source contributions are a positive signal. Discussing that fact is a bit out of scope for this FAQ anyway.